xbmc/video/jobs/VideoLibraryRefreshingJob.cpp

```diff
@@ -72,11 +76,117 @@ bool CVideoLibraryRefreshingJob::Work(CVideoDatabase &db)
 
   // determine the scraper for the item's path
   VIDEO::SScanSettings scanSettings;
-  ADDON::ScraperPtr scraper = db.GetScraperForPath(m_item->GetPath(), scanSettings);
+  ScraperPtr scraper = db.GetScraperForPath(m_item->GetPath(), scanSettings);
   if (scraper == nullptr)
-    return false;
+  {
+    if (m_item->GetVideoContentType() == VideoDbContentType::MOVIE_SETS)
+    {
+      // Deal with refreshing movie set
+      // Get movies in set
+      CVideoInfoTag tag{};
+      CFileItemList movies;
+      if (db.GetMoviesNav(m_item->GetPath(), movies, -1, -1, -1, -1, -1, -1,
+                          m_item->GetVideoInfoTag()->m_set.id, -1, SortDescription(),
+                          VideoDbDetailsAll))
+      {
+        // Scan each movie folder for an NFO file and get online information for a movie in the set
+        bool localFound = false;
+        bool onlineFound = false;
+
+        for (const auto& movie : movies)
+        {
+          // Check not plugin
+          if (!URIUtils::IsPlugin(movie->GetDynPath()))
+          {
+            // Use local scraper
+            AddonPtr addon{};
+            CServiceBroker::GetAddonMgr().GetAddon("metadata.local", addon,
+                                                   OnlyEnabled::CHOICE_YES);
+            scraper = std::dynamic_pointer_cast<CScraper>(addon);
+
+            // See if NFO
+            movie->SetPath(movie->GetDynPath());
+            CVideoTagLoaderNFO* nfo = new CVideoTagLoaderNFO(*movie, scraper, true);
```

Comment:
I don't see a matching `delete` probably better to use a `unique_ptr` anyways.

Reply:
Done. Thanks.